### PR TITLE
Fix panic threat when using type-cast for customClaims in validator

### PR DESCRIPTION
**Pull request by osamingo**
## Description

The threat of panicking should be curbed by receiving a second return value when executing the type-cast.

## References

- CodeReviewComments: https://github.com/golang/go/wiki/CodeReviewComments#dont-panic
- Effective Go: https://go.dev/doc/effective_go#interface_conversions

## Testing

- [x] This change adds test coverage for new/changed/fixed functionality

## Checklist

- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [x] I have reviewed my own code beforehand.
- [x] I have added documentation for new/changed functionality in this PR.
- [x] All active GitHub checks for tests, formatting, and security are passing.
- [ ] The correct base branch is being used, if not `master`.

### CODE REVIEW

**Comment by osamingo** [comment]
I would appreciate it if you would approve running GitHub Actions for this PR when you have time.

**Comment by osamingo** [comment]
@Widcket ping

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ import (
 )
 
 var handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-	claims := r.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
+	claims, _ := r.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
```
Comment (sergiught) [inline README.md:53]: These make sense in the examples, but I'd rather we don't omit checking for the ok here, as we're not really fixing the panic threat if the ok is unchecked.
```suggestion
claims, ok := r.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
	if !ok {
		http.Error(w, "failed to get validated claims", http.StatusInternalServerError)
		return
	}
```

File: examples/http-example/main.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func (c *CustomClaimsExample) Validate(ctx context.Context) error {
 }
 
 var handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-	claims := r.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
+	claims, _ := r.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
```
Comment (sergiught) [inline examples/http-example/main.go:31]: Same as above.

File: examples/http-jwks-example/main.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 var handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-	claims := r.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
+	claims, _ := r.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
```
Comment (sergiught) [inline examples/http-jwks-example/main.go:16]: Same as above.

File: validator/validator.go [hunk 4]
```diff
@@ -150,7 +150,7 @@ func (v *Validator) ValidateToken(ctx context.Context, tokenString string) (inte
 	}
 
 	if v.customClaims != nil {
-		validatedClaims.CustomClaims = claimDest[1].(CustomClaims)
+		validatedClaims.CustomClaims, _ = claimDest[1].(CustomClaims)
```
Comment (sergiught) [inline validator/validator.go:153]: This isn't actually necessary here, as we know for sure that `v.customClaims` is of type `func() CustomClaims` and on L117 we actually append that to `claimDest` if it's set. So we know for sure that on line 153, `claimDest[1]` is of type `CustomClaims`.

There is however one scenario where we could pass incorrectly a custom claim like this:

```go
	customClaims := func() validator.CustomClaims {
		return nil
	}
```

So what instead we could do is on L116 ensure we don't just check for `if v.customClaims != nil` but also `if v.customClaims() != nil`. 

Wdyt?

**Review by sergiught — COMMENTED** [review]
Thanks @osamingo for the contribution! I left a few comments.

File: validator/validator.go (hunk 4 above)
Comment (osamingo) [reply validator/validator.go:153]: @sergiught Sorry for the late reply 🙏🏼 

> So what instead we could do is on L116 ensure we don't just check for if v.customClaims != nil but also if v.customClaims() != nil.

I agree with your idea.
I would add this nil guard on L.116, and then in L.153, I would add a code comment to refer to L.116 👍🏼

**Comment by osamingo** [comment]
@sergiught I fixed it, PTAL.

**Comment by sergiught** [comment]
Hey @osamingo 👋🏻 thanks a lot for the changes and apologies it took so while to get back to you! Would you mind adding a test case for this as well in the `validatior_test`.go file? 🙏🏻

**Comment by sergiught** [comment]
Hey @osamingo 👋🏻 I had some time and went ahead and added the test cases missing. Thanks for the contribution again!

**Review by sergiught — APPROVED** [review]

**Pull request merged**